src/seedu/addressbook/commands/AddCommand.java

```diff
@@ -41,11 +44,23 @@ public AddCommand(String name,
                       String phone, boolean isPhonePrivate,
                       String email, boolean isEmailPrivate,
                       String address, boolean isAddressPrivate,
+                      String postalCode, boolean isPostalCodePrivate,
                       Set<String> tags) throws IllegalValueException {
         final Set<Tag> tagSet = new HashSet<>();
         for (String tagName : tags) {
             tagSet.add(new Tag(tagName));
         }
+
+        PostalCode newPostalCode = new PostalCode(postalCode, isPostalCodePrivate);
+        try{
+            if(newPostalCode!=null){
+                address = newPostalCode.retrieveMatchingAddress();
+            }
+        }
+        catch(IOException e){
```

Comment:
The form of try-catch block is not correct. Please see https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#layout (point 11)